### ISSUE TITLE
[BUG] layout prop breaks MotionValue 

### ISSUE THREAD

**Issue by bakoushin** [framer/motion#757]
If an element has a `layout` prop, `MotionValue` is not updated while dragging that element, so dependent elements don't animate.

[CodeSandbox with reproduction](https://codesandbox.io/s/framer-motion-path-drawing-drag-and-usetransform-forked-f2uk9?file=/src/Example.tsx:850-856) – based on the official Motion Value Example Sandbox

**Steps to reproduce**

1. Use official Motion Value Example Sandbox with`layout` prop added to the `box` element (link above)
2. Drag element left and right
4. Notice that the background doesn't animate, nor does the icon inside the box.

**Expected behavior**

Adding `layout` prop doesn't break `MotionValue` behavior.

**Comment by InventingWithMonster** [comment]
This is currently expected as layout + drag uses a calculated transform rather than working on the x/y values themselves. You would expect to receive weird output from this. 

The onViewportBoxUpdate prop callback provides two arguments, the current viewport box and the delta. This delta is sort of a fake one we calculate just for this callback, one that makes more sense for humans than the calculated transform. This can be used to get read-only access to the x/y you’re probably looking for.

**Comment by bakoushin** [comment]
Thanks for this explanation!

So, it is intended that `layout` prop prevents `MotionValue` from updating _automagically_. But we still can manually update the `MotionValue` by explicitly setting it value in `onViewportBoxUpdate`. 

For instance, if we want to track `x` movement:
```javascript
const x = useMotionValue(0);

return (
  <motion.div
    layout
    drag="x"
    onViewportBoxUpdate={(_, delta) => x.set(delta.x.translate)}
    ...
```
Then we can use `x` as usual in transforms etc whatever depends on `x` position.

Probably it is worth adding a note on this in `layout` prop [docs](https://www.framer.com/api/motion/component/#layoutprops.layout), what do you think?

**Comment by bakoushin** [comment]
Another issue with detached `MotionValue`.

Can we stop layout animation at any given point of time like we could with [`MotionValue.stop()`](https://www.framer.com/api/motion/motionvalue/#motionvalue.stop)?

For example: I want to achieve a "swipe-to-remove" animation in a list of items.

In this [CodeSanbox example](https://codesandbox.io/s/framer-motion-notifications-forked-uscb8?file=/src/Example.tsx) (based on official Notifications Demo) each notification could be removed from the stack by dragging to the left.

Notice, that before starting an `exit` animation, a box **first bounces back near to its initial position**.
Is there a way to stop it from bouncing back and just start `exit` animation from where the drag gesture ended?

Metaphorically speaking, how can we cut the invisible spring holding it? It is achievable using `MotionValue.stop()`, but since `layout` detaches `MotionValue` from the element, it's no longer an option.

Maybe there are other ways to achieve this?

**Comment by johnnybenson** [comment]
I think I have a use case that's encountering issues related to `layout` detaching from `MotionValue` but different enough that it's worth sharing.

Using `layout` to move video elements from portrait to landscape, I would like to use the scaleX and scaleY `MotionValue` to help video keep its aspect ratios during the transition. Scale correction and `object-fit` are two ships sailing passed each other in the night. 

I could be heading in the wrong direction, but I put together a sandbox to show what I mean, the video is squished during the transition and needs a little extra help.

I _think_ if I could apply parent scaleX to the child scaleY and parent scaleY to child scaleX moving in either direction, I could help the video keeps it's aspect ratio.

https://codesandbox.io/s/framer-motion-2-animate-scale-correction-with-video-jvet9

Is there a better way to do this?

**Comment by irg1008** [comment]
What happened? Did you managed to solve it?

**Comment by irg1008** [comment]
What happened? Were you able to solve it?

**Comment by mattgperry** [comment]
This has been solved in the meantime
